treon/test_execution.py

```diff
@@ -1,16 +1,32 @@
+import logging
 import os
 import textwrap
 import nbformat
 
 from nbformat.v4 import new_code_cell
-from nbconvert.preprocessors import ExecutePreprocessor
+from nbconvert.preprocessors import ExecutePreprocessor, CellExecutionError
 
+LOG = logging.getLogger('treon')
 
-def execute_notebook(path):
+def execute_notebook(path, html):
     notebook = nbformat.read(path, as_version=4)
     notebook.cells.extend([unittest_cell(), doctest_cell()])
     processor = ExecutePreprocessor(timeout=-1, kernel_name='python3')
-    processor.preprocess(notebook, metadata(path))
+    try:
+        processor.preprocess(notebook, metadata(path))
+    except CellExecutionError:
+        if html:
```

Comment:
It seems like html will only be generated if there's execution error. If there's an HTML flag then I'm guessing user is expecting HTML output even if the notebook executed successfully.